pytorch_lightning/strategies/ipu.py

```diff
@@ -214,10 +214,11 @@ def lightning_module(self) -> Optional["pl.LightningModule"]:
     def _convert_to_poptorch_loader(
         self, dataloader: DataLoader, sampler, mode: Optional[RunningStage] = None
     ) -> "poptorch.DataLoader":
-        dl_kwargs = _get_dataloader_init_kwargs(dataloader, sampler)
-        # Override to drop last uneven batch, as IPUs does not support uneven inputs.
-        dl_kwargs["drop_last"] = True
```

Comment:
I'm unsure why we were dropping this key, they do support it:

https://docs.graphcore.ai/projects/poptorch-user-guide/en/latest/reference.html#poptorch.DataLoader

And drop_last does not mean uneven inputs, but whether we drop the uneven ones or pad with repeated samples